### ISSUE TITLE
Problems with limited charging infrastructure when there is heavy electric ride hail use

### ISSUE THREAD

**Issue by estandiaa-marain** [LBNL-UCB-STI/beam#2225]
Hello team,
I am running a scenario with heavy electric ride hail use and limited charging infrastructure. The simulation completes but throws many different errors and warnings. The simulation is based on sf-light-25k with some modifications (see attached [config](https://github.com/LBNL-UCB-STI/beam/files/3726693/sf-light-25k.conf.txt) and [files](https://github.com/LBNL-UCB-STI/beam/files/3726690/dataToReproduce.zip)).  I am also attaching the [log](https://github.com/LBNL-UCB-STI/beam/files/3726692/beamLog.out.txt).

Looking at the log, I saw that there are some issues that happen multiple times during the simulation:

```
19:08:27.898 [ClusterSystem-akka.actor.default-dispatcher-170] WARN  beam.agentsim.agents.PersonAgent - Tick:36318 TriggId:1100698 State:PassengerScheduleEmpty PersonAgent:026003-2013000475568-0-1329929  Expected currentBeamVehicle.stall to be defined.
```

```
19:19:26.570 [ClusterSystem-akka.actor.default-dispatcher-244] ERROR beam.agentsim.agents.PersonAgent - Vehicle has no stall while ending charging event
```

```
19:14:18.958 [ClusterSystem-akka.actor.default-dispatcher-201] ERROR akka.actor.OneForOneStrategy - None.get
java.util.NoSuchElementException: None.get
	at scala.None$.get(Option.scala:529)
	at scala.None$.get(Option.scala:527)
	at beam.agentsim.agents.modalbehaviors.DrivesVehicle.handleEndCharging(DrivesVehicle.scala:902)
	at beam.agentsim.agents.modalbehaviors.DrivesVehicle.handleEndCharging$(DrivesVehicle.scala:874)
	at beam.agentsim.agents.PersonAgent.handleEndCharging(PersonAgent.scala:233)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.$anonfun$applyOrElse$3(PersonAgent.scala:448)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.$anonfun$applyOrElse$3$adapted(PersonAgent.scala:446)
	at scala.Option.foreach(Option.scala:407)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.$anonfun$applyOrElse$2(PersonAgent.scala:446)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.$anonfun$applyOrElse$2$adapted(PersonAgent.scala:443)
	at scala.collection.mutable.HashMap.$anonfun$foreach$1(HashMap.scala:149)
	at scala.collection.mutable.HashTable.foreachEntry(HashTable.scala:237)
	at scala.collection.mutable.HashTable.foreachEntry$(HashTable.scala:230)
	at scala.collection.mutable.HashMap.foreachEntry(HashMap.scala:44)
	at scala.collection.mutable.HashMap.foreach(HashMap.scala:149)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.applyOrElse(PersonAgent.scala:443)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.applyOrElse(PersonAgent.scala:435)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
	at akka.actor.FSM.processEvent(FSM.scala:706)
	at akka.actor.FSM.processEvent$(FSM.scala:703)
	at beam.agentsim.agents.PersonAgent.akka$actor$LoggingFSM$$super$processEvent(PersonAgent.scala:233)
	at akka.actor.LoggingFSM.processEvent(FSM.scala:846)
	at akka.actor.LoggingFSM.processEvent$(FSM.scala:828)
	at beam.agentsim.agents.PersonAgent.processEvent(PersonAgent.scala:233)
	at akka.actor.FSM.akka$actor$FSM$$processMsg(FSM.scala:700)
	at akka.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:694)
	at akka.actor.Actor.aroundReceive(Actor.scala:539)
	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
	at beam.agentsim.agents.PersonAgent.aroundReceive(PersonAgent.scala:233)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
19:14:18.959 [ClusterSystem-akka.actor.default-dispatcher-201] ERROR beam.agentsim.agents.PersonAgent - BeamAgent Got Shutdown. This means actorRef.stop() was called externally, e.g. by supervisor because of an exception. In state PerformingActivity
```

```
19:18:55.875 [ClusterSystem-akka.actor.default-dispatcher-239] ERROR akka.actor.OneForOneStrategy - None.get
java.util.NoSuchElementException: None.get
	at scala.None$.get(Option.scala:529)
	at scala.None$.get(Option.scala:527)
	at beam.agentsim.agents.modalbehaviors.DrivesVehicle.handleEndCharging(DrivesVehicle.scala:897)
	at beam.agentsim.agents.modalbehaviors.DrivesVehicle.handleEndCharging$(DrivesVehicle.scala:874)
	at beam.agentsim.agents.PersonAgent.handleEndCharging(PersonAgent.scala:233)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.$anonfun$applyOrElse$3(PersonAgent.scala:448)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.$anonfun$applyOrElse$3$adapted(PersonAgent.scala:446)
	at scala.Option.foreach(Option.scala:407)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.$anonfun$applyOrElse$2(PersonAgent.scala:446)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.$anonfun$applyOrElse$2$adapted(PersonAgent.scala:443)
	at scala.collection.mutable.HashMap.$anonfun$foreach$1(HashMap.scala:149)
	at scala.collection.mutable.HashTable.foreachEntry(HashTable.scala:237)
	at scala.collection.mutable.HashTable.foreachEntry$(HashTable.scala:230)
	at scala.collection.mutable.HashMap.foreachEntry(HashMap.scala:44)
	at scala.collection.mutable.HashMap.foreach(HashMap.scala:149)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.applyOrElse(PersonAgent.scala:443)
	at beam.agentsim.agents.PersonAgent$$anonfun$3.applyOrElse(PersonAgent.scala:435)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
	at akka.actor.FSM.processEvent(FSM.scala:706)
	at akka.actor.FSM.processEvent$(FSM.scala:703)
	at beam.agentsim.agents.PersonAgent.akka$actor$LoggingFSM$$super$processEvent(PersonAgent.scala:233)
	at akka.actor.LoggingFSM.processEvent(FSM.scala:846)
	at akka.actor.LoggingFSM.processEvent$(FSM.scala:828)
	at beam.agentsim.agents.PersonAgent.processEvent(PersonAgent.scala:233)
	at akka.actor.FSM.akka$actor$FSM$$processMsg(FSM.scala:700)
	at akka.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:694)
	at akka.actor.Actor.aroundReceive(Actor.scala:539)
	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
	at beam.agentsim.agents.PersonAgent.aroundReceive(PersonAgent.scala:233)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
19:18:55.876 [ClusterSystem-akka.actor.default-dispatcher-239] ERROR beam.agentsim.agents.PersonAgent - BeamAgent Got Shutdown. This means actorRef.stop() was called externally, e.g. by supervisor because of an exception. In state PerformingActivity
```

```
19:27:58.066 [ClusterSystem-akka.actor.default-dispatcher-284] WARN  beam.agentsim.agents.PersonAgent - [32768] Performing Activity at end of simulation
```

**Comment by wrashid** [comment]
- around 0.03% agents gone to error with exceptions reported above
- the key not found: walk_transit exception in log is resolved already

**Comment by wrashid** [comment]
```03:17:13.901 ERROR b.a.a.r.RideHailVehicleManager - Vehicle rideHailVehicle-108 does not have a service status, assuming out of service
java.util.NoSuchElementException: key not found: rideHailVehicle-108```

**Comment by JustinPihony** [comment]
Fixed by #2374
